### ISSUE TITLE
Port number in url is not supported?

### ISSUE THREAD

**Issue by koko236** [giginet/Crossroad#34]
Hi, 

is crossroads supposed to handle URLs with port numbers eg. 127.0.0.1:8080? 

I'm debugging an app where routes are not handled properly and it's seems that if I register routes to a router initialised like so:
```
let router = Router<Void>(url:  URL(from: "http://bloblo")!); // <<< Note no port number.
router.register([
            ("/about", { context in
                self.openWebPage(url: "/about");
                return true;
            })
])
```
Then router will respond correctly to `http://bloblo/about`.

BUT if I register them to a router initialised like so:
```
Router<Void>(url:  URL(from: "http://bloblo:8080")!); // <<< Note the port number.
```
Then router will not respond to `http://bloblo:8080/about`.

**Comment by giginet** [comment]
@koko236 
Thanks! Sorry to keep you waiting. 🙏 

Crossroad treats `:` to indicate variable names. So our DSL doesn't support port on URL.

In my understood, we needn't treat port numbers in most cases.

For debugging, you can use [custom URL schemes](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app) instead.

Could you explain the situation when URL schemes or universal links include port numbers?
Thank you.

**Comment by giginet** [comment]
@koko236
I released version 4.0.0. 🎉 

This version respects RFC validated URL it includes port numbers.
https://github.com/giginet/Crossroad/blob/4.0.0/Tests/CrossroadTests/PatternTests.swift#L38-L42

Could you try this version?
